techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st

```diff
@@ -112,83 +107,77 @@ classes:
       &GENERIC_FILE_CONTENT_ONLY_ZONE_BOOLEAN:{only_zone |"edit_only_in_a_zone_&i&" expression => strcmp("&only_zone&", "true");
 }&
 
-      "edit_section_valid_${index}" and => { isvariable("generic_file_content_section_header[${index}]"), isvariable("generic_file_content_section_footer[${index}]") } ;
+      "header_defined_${index}" not => strcmp("${generic_file_content_section_header[${index}]}", "");
+      "footer_defined_${index}" not => strcmp("${generic_file_content_section_footer[${index}]}", "");
+      "edit_section_valid_${index}" expression => "header_defined_${index}.footer_defined_${index}";
+
 
       "file_doesnt_exist_${index}"                       not => fileexists("${generic_file_content_path[${index}]}");
       "creation_only_but_already_exists_${index}" expression => "enforce_at_creation_only_${index}.!file_doesnt_exist_${index}";
 
       # define if we need to migrate to zone
       "migrate_to_zone_${index}" expression => strcmp("${migrate_to_zone[${index}]}", "true");
 
-      # Workaround for #3014: no value = empty value
-      # In some conditions, the value of replacement is
-      # considerered as null instead of empty.
-      # This class will ensure that no variable (=null) will be considered
-      # as an empty variable.
-      "generic_file_content_modification_destination_defined_${index}" expression => isvariable("generic_file_content_modification_destination[${index}]");
       # Check that value has been set before to edit the file
-      "edit_content_${index}" expression => isvariable("generic_file_content_payload[${index}]");
+      "generic_file_content_modification_destination_defined_${index}" not => strcmp("${generic_file_content_modification_destination[${index}]}", "");
+      "edit_content_${index}" not => strcmp("${generic_file_content_payload[${index}]}", "");
 
-    tier1::
+    pass1::
       # if the file absent and we don't plan to create it, then there is nothing to do on it
       "file_nothing_to_do_${index}" expression => strcmp("${generic_file_content_creation[${index}]}", "false"),
         ifvarclass => "file_doesnt_exist_${index}";
 
-    any::
-      "tier1" expression => "any";
-      "tier2" expression => isvariable("file_edit_go");
```

Comment:
Do you know why this was done this way?

Reply:
it was before the standart pass1/pass2/pass3 method